### DISCUSSION TITLE
Is terragrunt are supported call child local variable from template.

### DISCUSSION THREAD

**Discussion by Navapon** [gruntwork-io/terragrunt#3673] [General]
The idea is,

We currently use terragrunt templates and want to call local variables from the child file.

Do you have any solution that the template can get tags local variable from a child ( calling template file )

Please give me the link, explanation, or example for this solution.

### Structure 

```bash
├── _templates
|  └── s3.hcl
├── resource
|  └── s3
|     └── terragrunt.hcl
└── terragrunt.hcl
```

### Template 
Path `_templates/s3.yaml`

```hcl
locals {
  tg_dir_path = get_terragrunt_dir()

  tags = {
    DataClassification = "" 
  }
}

inputs = {
  is_kms_enabled = local.tags.DataClassification == 'internal' ? false : true
}
```

The caller template is like

### Child
```hcl

include "template" {
  path                    = "${get_repo_root()}/landing-zone/_templates/sample-template.yaml"
  merge_strategy = "deep"
  expose                = true
}

include "root" {
  path                     = find_in_parent_folders()
  merge_strategy = "deep"
  expose                 = true
}

locals {
   tags = {
    DataClassification = "internal" 
  }
}

inputs = {

}
```

**Comment by yhakbar** [comment]
Hey @Navapon , check this out: https://terragrunt.gruntwork.io/docs/features/includes/#using-read_terragrunt_config

**Reply by Arpo201** [reply]
[read_terragrunt_config](https://terragrunt.gruntwork.io/docs/features/includes/#using-read_terragrunt_config) can't use here. Because `child` include `template`, if `template` read_terragrunt_config `child`, it will be cause of loop.

```hcl
├── _templates
|  └── s3.hcl
├── resource
|  └── s3
|     └── terragrunt.hcl
└── terragrunt.hcl
```

**Reply by yhakbar** [reply]
The pattern that most folks use with Terragrunt is to organize infrastructure into a hierarchy, then to look up for any data.

For example, say you have some internal buckets, and some external buckets. You might do something like the following to set that up:

```bash
$ eza -T
.
├── _templates
│   └── s3.hcl
├── live
│   └── storage
│       ├── external
│       │   ├── classification.hcl
│       │   └── first-external
│       │       ├── main.tf
│       │       └── terragrunt.hcl
│       └── internal
│           ├── classification.hcl
│           └── first-internal
│               ├── main.tf
│               └── terragrunt.hcl
└── root.hcl
```

Then your HCL files would look like this:

```bash
$ fd -tf -x bash -c 'echo "# {}" && cat {} && echo'
# ./live/storage/external/first-external/main.tf

# ./live/storage/external/first-external/terragrunt.hcl
include "root" {
  path = find_in_parent_folders("root.hcl")
}

include "template" {
  path = "${dirname(find_in_parent_folders("root.hcl"))}/_templates/s3.hcl"
}

# ./_templates/s3.hcl
locals {
  classification = read_terragrunt_config(find_in_parent_folders("classification.hcl")).locals.classification
}

inputs = {
  is_kms_enabled = local.classification == "internal" ? false : true
}

# ./live/storage/internal/first-internal/terragrunt.hcl
include "root" {
  path = find_in_parent_folders("root.hcl")
}

include "template" {
  path = "${dirname(find_in_parent_folders("root.hcl"))}/_templates/s3.hcl"
}

# ./live/storage/internal/classification.hcl
locals {
  classification = "internal"
}

# ./live/storage/internal/first-internal/main.tf

# ./root.hcl

# ./live/storage/external/classification.hcl
locals {
  classification = "external"
}
```

If you haven't done it, I recommend going through the [Getting Started guide](https://terragrunt.gruntwork.io/docs/#getting-started). It can help you learn about the standard patterns folks typically use when setting up Terragrunt repositories.

**Comment by Navapon** [comment]
I found out that the solution that I am using calls the child variable at template area by using

```
run_cmd("render-json cli option")
```

To get the child value and add condition variable, that need for template
